srml/transaction-payment/src/lib.rs

```diff
@@ -97,7 +98,39 @@ decl_module! {
 	}
 }
 
-impl<T: Trait> Module<T> {}
+impl<T: Trait> Module<T> {
+	/// Query the data that we know about the fee of a given `call`.
+	///
+	/// As this module is not and cannot be aware of the internals of a signed extension. It only
+	/// interprets them as some encoded value and takes their length into account.
+	///
+	/// All dispatchables must be annotated with weight and will have some fee info. This function
+	/// always returns.
+	// TODO: we can actually make it understand `ChargeTransactionPayment`, but would be some hassle
+	// for sure. We have to make it aware of the index of `ChargeTransactionPayment` in `Extra`.
+	pub fn query_info_unsigned<Extra: Codec>(
+		sender: T::AccountId,
+		call: T::Call,
+		extra: Extra
+	) -> RuntimeDispatchInfo<BalanceOf<T>>
+		where T::Call: Codec + GetDispatchInfo, T: Send + Sync
+	{
+		// TODO: We can not decode this stuff and receive some raw bytes from the RPC, save the
+		// lengths, then decode and check that they are decodable (aka. correct) and just use the
+		// values.
+		let encoded_extra = extra.encode();
+		let encoded_call = call.encode();
+		let encoded_sender = sender.encode();
+
+		let dispatch_info = <T::Call as GetDispatchInfo>::get_dispatch_info(&call);
+		let len = encoded_call.len() + encoded_extra.len() + encoded_sender.len() + 64;
```

Comment:
What's the magic `64` for?

Reply:
This should go away with raw - since the signature is not provided, it assumed that is is the 64 bytes from `Signature` (but also needs the actual signing flags, i.e. `0x83` which is present in the raw extrinsic as well)

So this magic will disappear.

Reply:
no longer there.